crml/sylo/src/directory/rpc/src/lib.rs

```diff
@@ -49,19 +50,26 @@ impl From<Error> for i64 {
 	}
 }
 
-impl<C, Block, Balance, AccountId> SyloDirectoryApi<Balance, AccountId> for SyloDirectory<C, Block>
+impl<C, Block, AccountId> SyloDirectoryApi<AccountId> for SyloDirectory<C, Block>
 where
 	Block: BlockT,
 	C: Send + Sync + 'static + ProvideRuntimeApi<Block> + HeaderBackend<Block>,
-	C::Api: SyloDirectoryRuntimeApi<Block, Balance, AccountId>,
-	Balance: Codec + BaseArithmetic,
+	C::Api: SyloDirectoryRuntimeApi<Block, AccountId>,
 	AccountId: Codec,
 {
-	fn scan(&self, point: Balance) -> Result<AccountId> {
+	fn scan(&self, point: U256) -> Result<AccountId> {
 		let api = self.client.runtime_api();
 		let best = self.client.info().best_hash;
 		let at = BlockId::hash(best);
 
+		// Accepts u256 but should only work on u128
+		if point > u128::MAX.into() {
+			return Err(RpcError {
+				code: ErrorCode:: ServerError(Error::Runtime.into()),
+				message: "Value too large.".into(),
+				data: Some(format!("Expected a 128bit integer but received {:?} instead", point).into()), 
+			})
+		}
 		let result = api.scan(&at, point).map_err(|e| RpcError {
```

Comment:
Could we just change this to use u256? It seems weird to use `Balance` for this function parameter. We might also run into issues if `Balance` type changes.

If not, can we please add some comments as to why we have to cast and possibly refer to an issue

Reply:
> Could we just change this to use u256? It seems weird to use Balance for this function parameter. We might also run into issues if Balance type changes.

@wqsz7xn This is more your domain.

> If not, can we please add some comments as to why we have to cast and possibly refer to an issue

Will do!

Reply:
https://github.com/dn3010/sylo-ethereum-contracts/blob/42b047841c2b6607bd97e6ef14a4f159226dca49/contracts/Directory.sol#L222-L254
You're right it's pretty weird.
Ethereum 'balances' are just uint256. This will never change. Cennznet's balance types can change as they are just polymorphically constrained to some pretty basic traits. This means if I want to be able to select a node I have to use an equivalent 'balance' types in Cennznet and can't just keep it as a concrete type.

Reply:
@stwiname I added some more comments but unfortunately I can't find any links/record of the issue Jordan mentioned.